### DISCUSSION TITLE
ModelQueryBuilder.macro 'getCount'

### DISCUSSION THREAD

**Discussion by sergefabre** [adonisjs/core#2961] [Help]
Hello
In the Doc Adonis V5 : https://docs.adonisjs.com/guides/database/query-builder#extending-modelquerybuilder the total is returned with 
`const result = await this.count('* as total')
      return result[0].total`
but, it's not working.
It's working with "$extrat": 
`const result = await this.count('* as total')
      return result[0].$extras.total`

Is it OK or is it a problem in my config ?

Thanks for the help

**Comment by arthurcvm** [comment]
Hi!
You set [this step](https://docs.adonisjs.com/guides/database/query-builder#informing-typescript-about-the-method)?
If you get the same problem, you can try use the `@computed` decorator to get $extras data like this:

```ts
@computed()
  public get getCount (): number {
    return this.$extras.total
  }
```

the reference of computed in docs:
https://docs.adonisjs.com/reference/orm/decorators#computed

**Reply by sergefabre** [reply]
Yes, [This step](https://docs.adonisjs.com/guides/database/query-builder#informing-typescript-about-the-method) works fine
Thanks for your help

**Comment by thetutlage** [comment]
Yup. I think the docs needs to be fixed in this case. Feel free to send a PR

**Reply by arthurcvm** [reply]
@thetutlage the computed decorator can is used in PR or no?

**Reply by thetutlage** [reply]
I think the macro should showcase the usage without the need of having a computed property.

**Reply by sergefabre** [reply]
@thetutlage Feel free to send a PR. Sorry, I am à novice on github... "send a PR" ?
